node/hubController.js

```diff
@@ -0,0 +1,222 @@
+     */
+    onboard(hubId, onboardingInfo) {
+        console.log("----------------- onboard");
+        return this._getHubInfo(hubId).then((hubInfo) => {
+            // do the onboarding and return token
+            var Onboarding = require(hubInfo.onboarding);
+            var onboarding = new Onboarding();
+            return onboarding.onboard(onboardingInfo);
+        }).catch((err) => {
+            this._logError(err, "onboard");
+        });
+    }
+
+    /**
+     * given the specific hub id, returns all the platforms which are connected to it
+     */
+    platforms(hubId, authInfo) {
+        console.log("----------------- platforms");
+        // will return hub getPlatform contents
+        return this._getHubInfo(hubId).then((hubInfo) => {
+            return this._createTranslator(hubInfo.translator, authInfo).then((hubInstance) => {
+                // hug get
+                return this._invokeMethod(hubInstance, authInfo, "getPlatforms", [true]);
+            });
+        }).catch((err) => {
+            this._logError(err, "platforms");
+        });
+    }
+
+    /**
+     * given the specific hub id and opent2tblob, returns the specific platform
+     */
+    getPlatform(hubId, authInfo, opent2tBlob) {
+        console.log("----------------- getPlatform");
+        return this._getHubInfo(hubId).then((hubInfo) => {
+            return this._createTranslator(hubInfo.translator, authInfo).then((hubInstance) => {
+                // platform get
+                var deviceInfo = {};
+                deviceInfo.hub = hubInstance;
+                deviceInfo.deviceInfo = {};
+                deviceInfo.deviceInfo.opent2t = opent2tBlob;
+                return this._invokeMethod(opent2tBlob.translator, deviceInfo, "get", [true]);
+            });
+        }).catch((err) => {
+            this._logError(err, "getPlatform");
+        });
+    }
+
+    /**
+     * given the specific hub id, opent2tblob, and resourceId, sets it with the given resourceBlob
+     */
+    setResource(hubId, authInfo, opent2tBlob, deviceId, resoureceId, resourceBlob) {
+        console.log("----------------- setResource");
+        return this._getHubInfo(hubId).then((hubInfo) => {
+            return this._createTranslator(hubInfo.translator, authInfo).then((hubInstance) => {
+                // resource set
+                var deviceInfo = {};
+                deviceInfo.hub = hubInstance;
+                deviceInfo.deviceInfo = {};
+                deviceInfo.deviceInfo.opent2t = opent2tBlob;
+
+                var propertyName = "postDevices" + this._capitalizeFirstLetter(resoureceId);
+
+                return this._setProperty(opent2tBlob.translator, deviceInfo, propertyName, deviceId, resourceBlob);
+            });
+        }).catch((err) => {
+            this._logError(err, "setResource");
+        });
+    }
+
+
+    /// *********************************
+    /// helper methods
+    /// *********************************
```

Comment:
Nit: can we stick to a single C# doc style commenting style for function headers similar to this one here?